### ISSUE TITLE
Need convenience command for removing news fragment files

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4315]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#4315|
|Type|enhancement|
|Created|2010-02-24 18:30:16Z|
|Branch|https://github.com/twisted/twisted/tree/delete-news-fragments-4315|

After running `build-news` as part of the ReleaseProcess and then reviewing and committing the changes, we need to delete the news fragment files (e.g. `2155.bugfix`).

Unfortunately, doing this correctly is a bit of a pain. My best guess...

`find . -name '*.bugfix' -or -name '*.misc' -or -name '*.feature' -or -name '*.removal' -or -name '*.doc' | xargs svn rm`

This should be simpler.

<details><summary>Searchable metadata</summary>

```
trac-id__4315 4315
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__totally_automated_release_infrastructure totally automated release infrastructure
branch__branches_delete_news_fragments_4315 branches/delete-news-fragments-4315
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__release_management release management
keywords__ 
time__1267036216000000 1267036216000000
changetime__1357209938000000 1357209938000000
version__None None
owner__therve therve
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml commented|
|-|-|

`NewsBuilder._headings.keys()` is a list of the actual file extensions that ought to be deleted.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

Should this be a new command (`clean-news`?) or be part of `build-news`?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

There doesn't seem to be any reason to make it a separate command.  Let's make it part of `build-news`.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_9"></a>|@therve commented|
|-|-|

(In [36557]) Branching to 'delete-news-fragments-4315'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve removed owner|
|-|-|

Alright, let's automate stuff some more! Please review.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_11"></a>|@tomprince commented|
|-|-|

[buildbot run](https://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/delete-news-fragments-4315)

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_12"></a>|@tomprince set owner to @therve|
|-|-|

1. The docstring should be updated to indicate that build-news must be run from an svn checkout, and that it deletes the news framgments from svn.
1. Before running 'svn rm', the presence of a svn repo should be checked (this code can be shared with ```buildAllTarballs```. I'm not sure what the appropriate behaviour is, if it isn't. I think it is probably to just delete the files. Thus, if it is run in a bzr or git repository, the commit will automatically pick up the removals.

Please resubmit for review.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_13"></a>|@therve removed owner|
|-|-|

Thanks for the review. I updated the docstring, and raised an error when it's not a SVN directory. The check is slightly different from the other one so I kept it separated. I don't think we should support other use cases for now.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_14"></a>|@tomprince set owner to @therve|
|-|-|

Looks good. Please commit.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_15"></a>|@therve set status to `closed`|
|-|-|

(In [36726]) Merge delete-news-fragments-4315

Author: therve
Reviewer: tom.prince
Fixes: [#4315](https://github.com/twisted/twisted/issues/4315)

Delete NEWS fragments when building release NEWS files.
